lib/librarian/puppet/util.rb

```diff
@@ -26,7 +26,21 @@ def rsync?
       # If the rsync configuration parameter is set, use rsync instead of FileUtils
       def cp_r(src, dest)
         if rsync?
-          Rsync.run(File.join(src, "/"), dest, ['-avz', '--delete'])
+          if Gem.win_platform?
+            src_clean = "#{src}".gsub(/^([a-z])\:/i,'/cygdrive/\1')
+            dest_clean = "#{dest}".gsub(/^([a-z])\:/i,'/cygdrive/\1')
+          else
+            src_clean = src
+            dest_clean = dest
+          end
+          debug { "Copying #{src_clean}/ to #{dest_clean}/ with rsync -avz --delete" }
+          result = Rsync.run(File.join(src_clean, "/"), File.join(dest_clean, "/"), ['-avz', '--delete'])
+          if result.success?
```

Comment:
This checks the result of the rsync. Ruby rsync module does not throw errors in case anything goes wrong, so we need extra checking.